### PR TITLE
MM-14927 Made the real SiteURL be used in incoming webhook URLs

### PR DESCRIPTION
**Pull request by levb**
#### Summary
Use config.SiteURL if available, rather than getSiteURL() which points at window.location rather than the configured SiteURL.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-14927

### CODE REVIEW

File: components/integrations/confirm_integration/confirm_integration.jsx [hunk 1]
```diff
@@ -117,8 +119,8 @@ export default class ConfirmIntegration extends React.Component {
                         }}
                     />
                     <CopyText
-                        idMessage='integrations.copy_client_secret'
-                        defaultMessage='Copy Client Secret'
+                        idMessage='integrations.copy_webhook_url'
+                        defaultMessage='Copy webhook URL'
```
Comment (jasonblais) [inline components/integrations/confirm_integration/confirm_integration.jsx]: FYI, I'm just asking from UX whether this tooltip text should be in title case or sentence case. Seems we're currently using both in the UI, but title case more often.

https://community.mattermost.com/core/pl/xpd6naptuidfdfb1n5yw8i1dah

Comment (levb) [reply components/integrations/confirm_integration/confirm_integration.jsx]: Sure, would you like me to change this now, for 5.11? I just caught that it was wrongly cut&pasted from a different link.

Comment (jfrerich) [reply components/integrations/confirm_integration/confirm_integration.jsx]: Can we move the hover to the RHS of the button so it doesn't block the text above?

image

Comment (jasonblais) [reply components/integrations/confirm_integration/confirm_integration.jsx]: The proposal from UX is to use `Copy webhook URL` as included in the PR. So no changes required.

@jfrerich Possibly, but that will be a larger change as the tooltip is used elsewhere in the UI, such as below. But I encourage you to post it in [UX Design channel](https://community.mattermost.com/core/channels/ux-design) for consideration

image

File: components/integrations/confirm_integration/confirm_integration.jsx [hunk 2]
```diff
@@ -58,6 +59,7 @@ export default class ConfirmIntegration extends React.Component {
         const outgoingHook = this.props.outgoingHooks[this.state.id];
         const oauthApp = this.props.oauthApps[this.state.id];
         const bot = this.props.bots[this.state.id];
+        const siteURL = this.props.siteURL || getSiteURL();
```
Comment (lieut-data) [inline components/integrations/confirm_integration/confirm_integration.jsx]: Maybe `getSiteURL()` is set in the `mapStateToProps` instead?

**Comment by levb** [comment]
@lieut-data I completely agree with the sentiment of "why we're doing this in just one case and not the others, e.g. team invite links". The thought had crossed my mind but I decided to do this fix narrowly for 5.11 because it was a specific issue, reported by a specific customer, in rare circumstances. TThe change in this specific case seemed uncontroversial, the other cases I did not understand sufficiently well.

From your comments it does appear that making broader changes is indeed something that we'd need to think through. So I encourage to perhaps file a separate ticket to review and fix the use of SiteURl in webapp.

File: components/integrations/confirm_integration/confirm_integration.jsx (hunk 2 above)
Comment (levb) [reply components/integrations/confirm_integration/confirm_integration.jsx]: Sure thing. You mean, as a function prop, or just move the `... || getSiteURL` to `mapStateProps`?

**Comment by lieut-data** [comment]
@levb, sounds good -- to be clear, does that mean deferring this change as well, given the potential inversion onto other customers who haven't configured SiteURL and go to configure an incoming webhook?

**Comment by levb** [comment]
> @levb, sounds good -- to be clear, does that mean deferring this change as well, given the potential inversion onto other customers who haven't configured SiteURL and go to configure an incoming webhook?

@lieut-data I can do a quick check of both values and pick one that is not localhost (or a 127.[subnet]), if any?

File: components/integrations/confirm_integration/confirm_integration.jsx (hunk 2 above)
Comment (lieut-data) [reply components/integrations/confirm_integration/confirm_integration.jsx]: Yeah, my proposal would be to move the `... || getSiteURL` to `mapStateToProps`, increasing the purity of the component as a measure of whether its output is fully defined by its input props.

**Comment by levb** [comment]
@lieut-data @jfrerich While addressing @lieut-data's feedback I found another place to fix, and a few more that we should not fix. Anyway, I think this should do it for 5.11. @jasonblais I missed the list view in the first pass.

**Comment by amyblais** [comment]
If this is important for v5.11, can you help prioritize reviews @lieut-data @jfrerich

**Comment by jasonblais** [comment]
After discussing with Jesse:

This can potentially be an unexpected change in behaviour for some customers, so I'm following up with customer support to more fully understand the issue and ensure this is the right step to make.

**Comment by jasonblais** [comment]
This is on hold for now until we hear back from the affected customer.

**Comment by levb** [comment]
After discussing with CS decided not to go forward with the change.

**Pull request closed without merging**